tips/TIP-0053/tip-0053.md

```diff
@@ -0,0 +1,82 @@
+---
+tip: 53
+title: Tagged Data Payload
+description: Block payload for arbitrary data
+author: Philipp Gackstatter (@PhilippGackstatter) <philipp.gackstatter@iota.org>
+discussions-to: https://github.com/iotaledger/tips/pull/156
+status: Draft
+type: Standards
+layer: Core
+created: 2023-11-02
+replaces: 23
+---
+
+# Summary
+
+This TIP is a slightly revised version of TIP-23 with updates to IOTA 2.0, like the type prefix of the _Tagged Data_
+Payload type and its syntactic validation. The payload concept offers a very flexible way to combine and encapsulate
+information in the IOTA protocol. This document proposes a basic payload type that allows the addition of arbitrary
+data.
+
+# Motivation
+
+The most flexible way to extend an existing object is by the addition of arbitrary data. This payload provides a way to
+do just that. An optional tag can be used to categorize the data.
+
+# Specification
+
+## Tagged Data
+
+The following table describes the serialization of a _Tagged Data Payload_ following the notation from
+[TIP-21](../TIP-0021/tip-0021.md):
+
+<details>
+  <summary>Tagged Data</summary>
+  <blockquote>Optional Data with an optional Tag.</blockquote>
+</details>
+<table>
+  <tr>
+    <td>
+      <b>Name</b>
+    </td>
+    <td>
+      <b>Type</b>
+    </td>
+    <td>
+      <b>Description</b>
+    </td>
+  </tr>
+  <tr>
+    <td>Payload Type</td>
+    <td>uint8</td>
+    <td>Set to <strong>value 0</strong> to denote a <i>Tagged Data Payload</i>.</td>
+  </tr>
+  <tr>
+    <td>Tag</td>
+    <td>(uint8)ByteArray</td>
+    <td>The tag of the data.</td>
+  </tr>
+  <tr>
+    <td>Data</td>
+    <td>(uint32)ByteArray</td>
+    <td>Binary data.</td>
+  </tr>
+</table>
+
+It is important to note that `Tag` is not considered by the protocol, it just serves as a marker for second layer
+applications.
+
+## Syntactic Validation
+
+- The length `Len` of the `Tag` field must satisfy: `0 <= Len <= 64`.
```

Comment:
out of interest: why this tag length limitation?

Reply:
I think it has historical reasons. I believe it was limited to `64` when tags were still indexed by the protocol (Chrysalis), presumably due to optimizing for a certain database implementation and it has simply carried over through Stardust and IOTA 2.0. We could allow a larger tag length as far as I'm aware, but no one has really asked for it.